llvm/lib/Target/AMDGPU/AMDGPUInsertDelayAlu.cpp

```diff
@@ -377,6 +413,17 @@ class AMDGPUInsertDelayAlu : public MachineFunctionPass {
             }
           }
         }
+
+        if (SII->isVALU(MI.getOpcode())) {
+          for (const auto &Op : MI.defs()) {
+            Register Reg = Op.getReg();
+            if (AMDGPU::isSGPR(Reg, TRI)) {
+              lastSGPRfromVALU = *(TRI->regunits(Reg).begin());
```

Comment:
```suggestion
              lastSGPRfromVALU = TRI->regunits(Reg).front();
```
Or use `[0]`.

Reply:
```suggestion
              lastSGPRfromVALU = *MCRegUnitIterator(Reg, TRI);;
```
Then you can just construct the iterator which will figure out the first element. That way you avoid make_range + begin.

Reply:
Clearly there are many ways to do this :)

Quite frankly, I think it's fine as is. Though the parenthesis are redundant, and it's more common in LLVM style to avoid redundant parenthesis.